target/linux/sunxi/patches-4.9/050-Add-Orangepi-PC-support.patch

```diff
@@ -0,0 +1,6784 @@
+ 	status = "disabled";
+ };
+ 
++&i2c0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&i2c0_pins>;
++	status = "okay";
++
++	eeprom_a: atmel,24c32@0x57 {	/* RTC module ZS-042 with eeprom */
++		compatible = "atmel,24c32";
++		reg = <0x00000057>;
++		pagesize = <32>;	/* appended from boardinfo */
++	}; //end atmel,24c32@0x57 (eeprom)
++
++	rtc_a: dallas,ds3231@0x68 {		/* RTC module ZS-042 with eeprom */
++		compatible = "dallas,ds3231";
++		reg = <0x00000068>;
++	}; //end dallas,ds3231@0x68 (rtc)
++};
++
++&i2c1 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&i2c1_pins>;
++	status = "okay";
++};
++
++&i2c2 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&i2c2_pins>;
++	status = "okay";
++};
++
++&spi0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&spi0_pins_a>;
++	status = "okay";
++
++	spidev@0 {
++		compatible = "allwinner,sun6i-a31-spi";
++		reg = <0>;
++		spi-max-frequency = <50000000>;
++	};
++};
++
++&usb_otg {
++	dr_mode = "otg";	/* or "peripheral" mode */
++	status = "okay";
++};
++
+ &usbphy {
+ 	/* USB VBUS is always on */
+-	status = "okay";
+-};
++	usb0_id_det-gpios = <&pio 6 12 GPIO_ACTIVE_HIGH>; /* PG12 */
++	usb0_vbus-supply = <&reg_usb0_vbus>;
++	status = "okay";
++};
++
++&emac {
++	phy-handle = <&int_mii_phy>;
++	phy-mode = "mii";
++	allwinner,leds-active-low;
++	status = "okay";
++};
+--- a/arch/arm/configs/multi_v7_defconfig
++++ b/arch/arm/configs/multi_v7_defconfig
+@@ -233,6 +233,8 @@
+ CONFIG_VIRTIO_NET=y
+ CONFIG_HIX5HD2_GMAC=y
+ CONFIG_SUN4I_EMAC=y
++CONFIG_SUN8I_EMAC=y
++CONFIG_SUN8I_THS=y
+ CONFIG_MACB=y
+ CONFIG_BCMGENET=m
+ CONFIG_SYSTEMPORT=m
+@@ -530,6 +532,7 @@
+ CONFIG_REGULATOR_QCOM_SMD_RPM=y
+ CONFIG_REGULATOR_S2MPS11=y
+ CONFIG_REGULATOR_S5M8767=y
++CONFIG_REGULATOR_SY8106A=y
+ CONFIG_REGULATOR_TI_ABB=y
+ CONFIG_REGULATOR_TPS51632=y
+ CONFIG_REGULATOR_TPS62360=y
+--- a/arch/arm/configs/sunxi_defconfig
++++ b/arch/arm/configs/sunxi_defconfig
+@@ -40,6 +40,8 @@
+ CONFIG_AHCI_SUNXI=y
+ CONFIG_NETDEVICES=y
+ CONFIG_SUN4I_EMAC=y
++CONFIG_SUN8I_EMAC=y
++CONFIG_SUN8I_THS=y
+ # CONFIG_NET_VENDOR_ARC is not set
+ # CONFIG_NET_CADENCE is not set
+ # CONFIG_NET_VENDOR_BROADCOM is not set
+@@ -93,6 +95,7 @@
+ CONFIG_REGULATOR_FIXED_VOLTAGE=y
+ CONFIG_REGULATOR_AXP20X=y
+ CONFIG_REGULATOR_GPIO=y
++CONFIG_REGULATOR_SY8106A=y
+ CONFIG_MEDIA_SUPPORT=y
+ CONFIG_MEDIA_RC_SUPPORT=y
+ CONFIG_RC_DEVICES=y
+--- a/drivers/net/ethernet/allwinner/Kconfig
++++ b/drivers/net/ethernet/allwinner/Kconfig
+@@ -33,4 +33,17 @@
+           To compile this driver as a module, choose M here.  The module
+           will be called sun4i-emac.
+ 
++config SUN8I_EMAC
++	tristate "Allwinner sun8i EMAC support"
++	depends on ARCH_SUNXI || COMPILE_TEST
++	depends on OF
++	select MII
++	select PHYLIB
++        ---help---
++	  This driver support the sun8i EMAC ethernet driver present on
++	  H3/A83T/A64 Allwinner SoCs.
++
++          To compile this driver as a module, choose M here.  The module
++          will be called sun8i-emac.
```

Comment:
We backported the upstream driver for the Ethernet core found in this SoC, please use that instead of adding a different driver.